test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java

```diff
@@ -139,6 +140,95 @@ public void insertNonPK()
         test.run();
     }
 
+    @Test
+    public void cannotInsertWrongNumberOfDimensionsFixedWidth() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int primary key, value vector<int, 2>)");
+
+        // fewer values than expected, with literals and bind markers
+        assertInvalidThrowMessage("Invalid vector literal for value of type vector<int, 2>; expected 2 elements, but given 1",
+                                  InvalidRequestException.class,
+                                  "INSERT INTO %s (pk, value) VALUES (0, [1])");
+        assertInvalidThrowMessage("Not enough bytes to read a vector<int, 2>",
+                                  InvalidRequestException.class,
+                                  "INSERT INTO %s (pk, value) VALUES (0, ?)", vector(1));
+
+        // more values than expected, with literals and bind markers
+        assertInvalidThrowMessage("Invalid vector literal for value of type vector<int, 2>; expected 2 elements, but given 3",
+                                  InvalidRequestException.class,
+                                  "INSERT INTO %s (pk, value) VALUES (0, [1, 2, 3])");
+        assertInvalidThrowMessage("Unexpected extraneous bytes after vector<int, 2> value",
+                                  InvalidRequestException.class,
+                                  "INSERT INTO %s (pk, value) VALUES (0, ?)", vector(1, 2, 3));
+    }
+
+    @Test
+    public void cannotInsertWrongNumberOfDimensionsVariableWidth() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int primary key, value vector<text, 2>)");
+
+        // fewer values than expected, with literals and bind markers
+        assertInvalidThrowMessage("Invalid vector literal for value of type vector<text, 2>; expected 2 elements, but given 1",
+                                  InvalidRequestException.class,
+                                  "INSERT INTO %s (pk, value) VALUES (0, ['a'])");
+        assertInvalidThrowMessage("Not enough bytes to read a vector<text, 2>",
+                                  InvalidRequestException.class,
+                                  "INSERT INTO %s (pk, value) VALUES (0, ?)", vector("a"));
+
+        // more values than expected, with literals and bind markers
+        assertInvalidThrowMessage("Invalid vector literal for value of type vector<text, 2>; expected 2 elements, but given 3",
+                                  InvalidRequestException.class,
+                                  "INSERT INTO %s (pk, value) VALUES (0, ['a', 'b', 'c'])");
+        assertInvalidThrowMessage("Unexpected extraneous bytes after vector<text, 2> value",
```

Comment:
what about show more detail messages about the left size as I have describe before ?
May be can be : "Unexpected extraneous bytes after vector<text, 2> value,  xxx size of text is given"